src/spi.rs

```diff
@@ -99,6 +99,15 @@ pub enum Error {
     _Extensible,
 }
 
+/// Enabled SPI peripheral (type state)
+pub struct Enabled;
+/// Disabled SPI peripheral (type state)
+pub struct Disabled;
+
+pub trait ED {}
+impl ED for Enabled {}
+impl ED for Disabled {}
```

Comment:
Is this trait really useful? (I know the other examples have it..)

It's only possible to instantiate `Spi<_,_,_>` in this module, so we can ensure that only `Enabled` and `Disabled` are used here. The generic `impl<EN>` below works just fine without the trait bound.

Then I think this trait can just be removed.